### ISSUE TITLE
pi-manage backup create doesn't work with mysql using ssl

### ISSUE THREAD

**Issue by gvde** [privacyidea/privacyidea#3833]
### Top-level intent

Trying to run

`
$ pi-manage backup create
`

### Steps to reproduce

1. In pi.cfg set ssl:
```
SQLALCHEMY_DATABASE_URI = 'mysql+pymysql://pi:pass@db.example.com/pi?ssl_ca=/etc/pki/tls/certs/root.pem'`
```
2. Set user in mysql to require ssl: 
```
GRANT USAGE ON *.* TO `pi`@`pi.example.com` IDENTIFIED BY PASSWORD '***' REQUIRE SSL
```
3. Run backup:
```
$ pi-manage backup create
...
mysqldump: Got error: 1045: "Access denied for user 'pi'@'pi.example.com' (using password: YES)" when trying to connect
```

### Expected outcome

It should recognized the ssl_ca parameter and establish the connection using SSL (i.e. pass at least `--ssl` to mysqldump)

### Configuration

* **privacyIDEA version**: 3.9.1
* **Installation method**: (from Ubuntu packages, github, PyPI, ...) PyPI
* **Python version**: 3.6.8
* **Operating system**: AlmaLinux 8.9
* **Webserver**: httpd
* **Token database**: (MySQL, PostgreSQL, ...) MySQL

**Comment by cornelinux** [comment]
You are trying to use a standard tool for a non-standard setup.

**Comment by gvde** [comment]
> You are trying to use a standard tool for a non-standard setup.

What is a non-standard setup? Using TLS for the database connection?

**Comment by cornelinux** [comment]
> Using TLS for the database connection?

Yes. The standard installation does not use your individual grant. 
I am not saying, this is bad or wrong. I am just saying, this is not a possible bug but a feature request.

**Comment by gvde** [comment]
Well, in the end it doesn't matter how you call it. For me it's a bug. The pi sql connection is configured in pi.cfg. It's in SQLALCHEMY_DATABASE_URI. It supports sqlalchemy options for mysql+pymysql which includes options to configure TLS.

The fact that pi-manager tries to set up my.cnf for mysqldump derived from SQLALCHEMY_DATABASE_URI and doesn't support TLS is from my user perspective a bug: everything is working fine with the configuration except for backup.

To me it's a design issue to use one syntax for the sql connection configuration and then having to map that syntax into another one for other tools. That can always cause issues...

**Comment by cornelinux** [comment]
You comment shows, that you do not understand how the sofware and the tools work. 
And in fact it matters a lot, how we call it. Because thereon the following actions depend.
But you are totally free to not care about how we call it.

**Comment by gvde** [comment]
> Closing issues due to bad energy.

Well, I suppose ignorance is bliss, if you are closing this, thus refusing to do anything about it regardless whether you call it "feature request" or not.

HA setups are documented https://privacyidea.readthedocs.io/en/v2.11/faq/high-availability.html which require a central database.

Use of SQLALCHEMY_DATABASE_URI is documented as well https://privacyidea.readthedocs.io/en/v2.11/faq/mysqldb.html#mysqldb

TLS is an option to PyMySQL.

Everything is documented in the privacyidea and not marked there as "non-standard" or "very special" or "not fully supported everywhere"...

I know how it works and I know why pi-manage backup doesn't work and I know why I need to trick pi-manage backup to work anyway. But I also read the documentation and I have an understanding what people know who read the documentation and what not and if it is written in the docs it's standard enough for me.
